pkg/expr/nodes.go

```diff
@@ -248,7 +248,7 @@ func (dn *DSNode) Execute(ctx context.Context, vars mathexp.Vars, s *Service) (m
 		if len(qr.Frames) == 1 {
 			frame := qr.Frames[0]
 			// Handle Untyped NoData
-			if len(frame.Fields) == 0 {
+			if len(frame.Fields) == 0 || frame.Fields[0].Len() == 0 {
```

Comment:
Perhaps it is different if there is 0 len fields, but say a time series that has 2 fields, and the numeric field has labels... In the case do we get a labeled instance? Do we want that to be no data?

Reply:
I think we want it to be no data, because the presence of labels does not mean there is data, unless specified in the data frame contract? However, perhaps we want to include the labels in the no data response?

Reply:
Ah, in the contract this is "empty", and it says in that case there should be the fields (time,number) but the field values are of zero length.

So I think this is okay actually - may interact with some datasource in an unintended way currently, but bravely forward! :-)

Reply:
Oh wait I take that back reading the issue again. I don't think we want it to be no data. As I would imagine Empty Series -> Reduce (with replace non-numeric option) should result in alerting or not-alerting depending on the replacement value. Is that correct?

Reply:
Although, with postgres/SQL/Tabular in particular, there shouldn't the possibility of labels if no rows. Because Long format (or the resulting Wide format from LongToWide) would not have dimensions (conceptually label)s if there are no rows to create them. But with other datasources it could be different